addons/sale/models/product_template.py

```diff
@@ -139,6 +139,11 @@ def _onchange_type(self):
             self.service_type = 'manual'
         return res
 
+    def write(self, values):
+        if 'type' in values and self.sales_count > 0:
```

Comment:
1) write is supposed to support multiple records edition `(len(self) > 1)`
`env['product.template'].browse([1,2,3]).write({'type': 'consu'})`
-> ERROR

2) writing the same value shouldn't trigger your warning
```python
product = env['product.product'].create({'type': 'consu'}
env['sale.order.line'].create({'product_id': product.id})
product.write({'type': 'consu'})
```

Reply:
Hello @Feyensv 
Thank you for the review, I agree with points 1 and 2, I fixed them.

However, I don't understand how the onchange solution can work, for example, if we do a RPC call? It is for this reason that I do the verification in the write

Reply:
I believe we can consider that modifications through rpc calls (without going through the web interface) are mainly used
by knowledgeable users, and they should be aware that this kind of change is quite risky.

As I see it, this warning is mostly a defensive safeguard against "dumb users making risky changes through the interface" (if the product was used already).   A simple onchange warning should be enough for those users, overrides of core orm methods should be avoided when not really necessary, especially on critical models frequently modified.

Furthermore, the onchange warning has the benefit of being triggered directly when the modification is done in the interface, not only on save (the sooner the warning/error, the less the user will make additional changes on the record).